src/main.rs

```diff
@@ -28,6 +28,44 @@ struct Cli {
     command: Commands,
 }
 
+#[derive(clap::Parser, ValueEnum, Debug, Clone, PartialEq, Eq)]
+enum CompletionShell {
+    Bash,
+    Elvish,
+    Fish,
+    Nushell,
+    PowerShell,
+    Zsh,
+}
+
+fn generate_main_shell(shell: clap_complete::Shell) {
+    generate(
+        shell,
+        &mut Cli::command(),
+        "starship",
+        &mut io::stdout().lock(),
+    )
+}
+fn generate_nushell(shell: clap_complete_nushell::Nushell) {
+    generate(
+        shell,
+        &mut Cli::command(),
+        "starship",
+        &mut io::stdout().lock(),
+    )
+}
```

Comment:
These can share a single definition:
```suggestion
fn generate_main_shell(shell: impl clap_complete::Generator) {
    generate(
        shell,
        &mut Cli::command(),
        "starship",
        &mut io::stdout().lock(),
    )
}
```